gammapy/modeling/models/temporal.py

```diff
@@ -617,6 +618,28 @@ def from_table(cls, table, filename=None):
 
         return cls(m, t_ref=t_ref, filename=filename)
 
+    @classmethod
+    def from_powerspectrum(cls, powerspectrum, map, **kwargs):
+        """Create a template model from a power spectrum using the Timmer algorithm.
+
+        Parameters
+        ----------
+        powerspectrum: `~numpy.ndarray`
```

Comment:
I think that Timmer Konig algorithm creates only power law periodograms, so allowing users to pass a random power spectrum is not safe. Rather take model parameters as input and generate the power spectrum internally

Reply:
Thanks for the comments @AtreyeeS, @bkhelifi.
 I don't think the algorithm itself is limited to power law noise. To report a paragraph at the end of section 2 of the paper _"For linear stochastic processes the spectrum and the periodogram are obtained by multiplying the results for the white noise process by the filter function (here: a power law) that describes the process. For these processes the variance of the complex random variable f(w) in eq. (9) becomes frequency dependent and is determined by the spectrum. Eq. (13) remains valid. This is also true for nonlinear stochastic processes and even for most chaotic processes."_
The _"here: a power law"_ comment is realized in the algorithm described by the paper by substituting sqrt(1/2 S(w_i)) with the power law case (1/w)^b/2 . I don't do this substitution in my implementation. 

Apart from that, I think the use of an array as input is helpful in the case of a periodogram derived from a dataset (for example using scipy or numpy) - so that the user doesn't have to pass through fitting. I feel that both ideas have their space - maybe as alternative inputs (either a set of parameters - or future "models", or an array/template?).

Reply:
But maybe I misinterpreted the passage of the paper I quoted. We can discuss it!

Reply:
@bkhelifi : I don't know the best way to handle time gaps in the data. During my PhD, I used to simulate time series with uniform time binning, and then apply masks as per the light curves being studied. Passing GTIs would emulate something along that line I imagine. Where the time series is internally generated without gaps and then the GTI selection is applied on the simulated data?

Reply:
Indeed, it seems the right approach for the simulation. In fact, I was referring to the API and the user workflow, for which I have not yet catched the whole logics...